pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -45,7 +49,72 @@ func (a Adapter) DoesComponentExist(cmpName string) bool {
 }
 
 // Delete attempts to delete the component with the specified labels, returning an error if it fails
-// Stub function until the functionality is added as part of https://github.com/openshift/odo/issues/2581
-func (d Adapter) Delete(labels map[string]string) error {
+func (a Adapter) Delete(labels map[string]string) error {
+
+	componentName, exists := labels["component"]
+	if !exists {
+		return errors.New("unable to delete component without a component label")
+	}
+
+	list, err := a.Client.GetContainerList()
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve container list for delete operation")
+	}
+
+	componentContainer := a.Client.GetContainersByComponent(componentName, list)
+
+	if len(componentContainer) == 0 {
+		return errors.Errorf("the component %s doesn't exist", a.ComponentName)
+	}
+
+	// Get all volumes that match our component label
+	volumeLabels := utils.GetProjectVolumeLabels(componentName)
+	vols, err := a.Client.GetVolumesByLabel(volumeLabels)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve source volume for component "+componentName)
+	}
+	if len(vols) == 0 {
+		return fmt.Errorf("unable to find source volume for component %s", componentName)
+	}
+
+	volumesToDelete := map[string] /* volume name -> volume name (value unused)*/ string{}
```

Comment:
forgot to remove the comment?

Reply:
It was intentional (I often use this format to document map key/value, eg `/* key description -> value description*/`), but I removed it and replaced it with a more standard description of the variable.